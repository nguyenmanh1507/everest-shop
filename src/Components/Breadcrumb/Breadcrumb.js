// @flow

import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { startCase, trimStart, split, uniqueId } from 'lodash'

import { GridContainer } from 'Components/Layouts'

type Props = {
  match: {
    url: string
  }
}

class Breadcrumb extends Component<Props> {
  render() {
    const { match: { url } } = this.props

    const newUrl = split(trimStart(url, '/'), '/')
    let breadcrumbUrl = ''

    return (
      <div className="bg-gray py-sm">
        <GridContainer exClass="cell">
          <div className="d-flex align-items-center">
            <Link
              to="/"
              className="text-muted mr-sm"
              data-test="bc-homepage-link"
            >
              <i className="fa fa-home" />
            </Link>
            {newUrl.map((u, index) => {
              if (index + 1 === newUrl.length) {
                return [
                  <i
                    className="fa fa-caret-right text-muted mr-sm"
                    key={uniqueId('breadcrumb-')}
                  />,
                  <span
                    className="text-small text-muted"
                    key={uniqueId('breadcrumb-')}
                    data-test="bc-item"
                  >
                    {startCase(u)}
                  </span>
                ]
              }

              breadcrumbUrl += `/${u}`

              return [
                <i
                  className="fa fa-caret-right text-muted mr-sm"
                  key={uniqueId('breadcrumb-')}
                />,
                <Link
                  to={breadcrumbUrl}
                  className="text-small mr-sm"
                  key={uniqueId('breadcrumb-')}
                  data-test="bc-item"
                >
                  {startCase(u)}
                </Link>
              ]
            })}
          </div>
        </GridContainer>
      </div>
    )
  }
}

export default Breadcrumb
