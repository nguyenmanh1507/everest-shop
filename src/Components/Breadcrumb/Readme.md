Breadcrumb example

```js
const { BrowserRouter } = require('react-router-dom');

<BrowserRouter>
  <Breadcrumb match={{url: '/women/clothing/t-shirts'}} />
</BrowserRouter>
```
