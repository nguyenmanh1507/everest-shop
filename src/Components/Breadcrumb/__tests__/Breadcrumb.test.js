// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'
import { render, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { trimStart, split } from 'lodash'

import Breadcrumb from '../Breadcrumb'

configure({ adapter: new Adapter() })

test('Breadcrumb render properly', () => {
  const match = { url: '/women/clothing/jeans' }

  const component = renderer.create(renderBreadcrumb(match))

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

describe('<Breadcrumb />', () => {
  const match = { url: '/women/clothing/jeans' }
  const wrapper = render(renderBreadcrumb(match))

  test("Always render homepage's link", () => {
    expect(wrapper.find('[data-test="bc-homepage-link"]')).toHaveLength(1)
  })

  test('Last item is span tag so it can not clicked', () => {
    expect(
      wrapper
        .find('[data-test="bc-item"]')
        .last()
        .is('span')
    ).toBeTruthy()
  })

  test('Items has same length with URL path', () => {
    const pathLength = split(trimStart(match.url, '/'), '/').length
    expect(wrapper.find('[data-test="bc-item"]')).toHaveLength(pathLength)
  })
})

function renderBreadcrumb(match) {
  const context = {}

  return (
    <StaticRouter context={context}>
      <Breadcrumb match={match} />
    </StaticRouter>
  )
}
