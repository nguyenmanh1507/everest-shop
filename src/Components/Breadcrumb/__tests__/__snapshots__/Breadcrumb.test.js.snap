// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb render properly 1`] = `
<div
  className="bg-gray py-sm"
>
  <div
    className="grid-container cell"
  >
    <div
      className="d-flex align-items-center"
    >
      <a
        className="text-muted mr-sm"
        data-test="bc-homepage-link"
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa fa-home"
        />
      </a>
      <i
        className="fa fa-caret-right text-muted mr-sm"
      />
      <a
        className="text-small mr-sm"
        data-test="bc-item"
        href="/women"
        onClick={[Function]}
      >
        Women
      </a>
      <i
        className="fa fa-caret-right text-muted mr-sm"
      />
      <a
        className="text-small mr-sm"
        data-test="bc-item"
        href="/women/clothing"
        onClick={[Function]}
      >
        Clothing
      </a>
      <i
        className="fa fa-caret-right text-muted mr-sm"
      />
      <span
        className="text-small text-muted"
        data-test="bc-item"
      >
        Jeans
      </span>
    </div>
  </div>
</div>
`;
